tipg/dbmodel.py

```diff
@@ -150,45 +160,47 @@ class Collection(BaseModel):
     dbschema: str = Field(..., alias="schema")
     title: Optional[str]
     description: Optional[str]
-    extent: Optional[Extent]
     properties: List[Column] = []
     id_column: Optional[str]
-    geometry_columns: List[GeometryColumn] = []
-    datetime_columns: List[DatetimeColumn] = []
-    geometry_column: Optional[GeometryColumn]
-    datetime_column: Optional[DatetimeColumn]
+    geometry_column: Optional[Column]
+    datetime_column: Optional[Column]
     parameters: List[Parameter] = []
     minzoom: int = tile_settings.default_minzoom
     maxzoom: int = tile_settings.default_maxzoom
     default_tms: str = tile_settings.default_tms
 
-    @root_validator
-    def extent_default(cls, values):
-        """Get default bounds from the first geometry columns."""
-        extent = {}
-
-        geoms = values.get("geometry_columns")
-        if geoms:
-            # Get the Extent of all the bounds
-            minx, miny, maxx, maxy = zip(*[geom.bounds for geom in geoms])
-            extent["spatial"] = {
-                "bbox": [[min(minx), min(miny), max(maxx), max(maxy)]],
-                "crs": f"http://www.opengis.net/def/crs/EPSG/0/{geoms[0].srid}",
-            }
+    @property
+    def geometry_columns(self):
+        """Return geometry columns."""
+        return [c for c in self.properties if c.is_geometry]
 
-        dates = values.get("datetime_columns")
-        if dates:
-            intervals = []
-            for d in dates:
-                if d.min or d.max:
-                    intervals.append([d.min, d.max])
-            if intervals:
-                extent["temporal"] = {"interval": intervals}
+    @property
+    def extent(self):
+        """Return extent."""
+        spatial = None
+        temporal = None
+        if self.geometry_column and self.geometry_column.bounds:
+            spatial = Spatial(bbox=[self.geometry_column.bounds], crs=self.crs)
+        if (
+            self.datetime_column
+            and self.datetime_column.mindt
+            and self.datetime_column.maxdt
+        ):
+            temporal = Temporal(
+                interval=[[self.datetime_column.mindt, self.datetime_column.maxdt]]
+            )
+        return Extent(spatial=spatial, temporal=temporal)
```

Comment:
We should get the extent from all the columns (`self.geometry_columns` and `self.datetime_columns`) not only form `self.datetime_column` and `self.geometry_column`

Reply:
in fact for spatial, the bbox should be in form of `[[total bounds], [bounds of geom col1], [bounds of geom col2], ....]

> One or more bounding boxes that describe the spatial extent of the dataset.      
         Extensions may support additional areas.
          The first bounding box describes the overall spatial
          extent of the data. All subsequent bounding boxes describe
          more precise bounding boxes, e.g., to identify clusters of data.
          Clients only interested in the overall spatial extent will
          only need to access the first bounding box in the array.

Reply:
Same for temporal 
> The first time interval describes the overall
temporal extent of the data. All subsequent time intervals describe
more precise time intervals, e.g., to identify clusters of data.
Clients only interested in the overall temporal extent will only need
to access the first time interval in the array (a pair of lower and upper
bound instants).

Reply:
I think getting the overall bounds should be priority for this PR. We can ticket getting the separate bounds for columns to another issue.

Reply:
see #21